src/screens/AddItem/AddItemNavigator.tsx

```diff
@@ -0,0 +1,31 @@
+import { NavigationContainer } from "@react-navigation/native";
+import { createNativeStackNavigator } from "@react-navigation/native-stack";
+import React from "react";
+import AddContainerScreen from "./AddContainerScreen";
+import AddItemScreen from "./AddItemScreen";
+import AddRoomScreen from "./AddRoomScreen";
+import SelectItemType from "./SelectItemType";
+
+const Stack = createNativeStackNavigator();
+interface AddItemNavigationProps {
+  navigation: any;
+  route: any;
+}
+
+export default function AddItemNavigation(props: AddItemNavigationProps) {
+  return (
+    <Stack.Navigator
+      initialRouteName="SelectItemType"
+      screenOptions={{ headerShown: false }}
+    >
+      <Stack.Screen name="SelectItemType" component={SelectItemType} />
+      <Stack.Screen
+        name="AddItemScreen"
+        component={AddItemScreen}
+        initialParams={props.route.params}
```

Comment:
This initial params is the only way I can find to make sure the room data always makes it to the create item screen.

Reply:
in the property interfaces you can define the route and params properties and this looks a lot less hacky now. That I believe is just the correct way to pass these parameters.